src/NHibernate.Validator/Engine/ClassValidator.cs

```diff
@@ -594,27 +764,35 @@ protected void ApplyToChildrenComponentsValidators(IEnumerable<Property> propert
 		{
 			foreach (var childGetter in childGetters)
 			{
-				Property property = properties.FirstOrDefault(p=> p.Name == childGetter.Name);
-				if (property != null && property.IsComposite && !property.BackRef)
+				var property = properties.FirstOrDefault(p=> p.Name == childGetter.Name);
+				if (property == null) 
+					continue;
+
+				if (property.IsComposite && !property.BackRef)
 				{
-					Component component = (Component) property.Value;
+					var component = property.Value as Component;
+					if (component == null) 
+						continue;

```

Comment:
It seems there are some "bug fix by the way". Was `Component component = (Component) property.Value;` failing in some cases? Has a test case been added for them?

Reply:
It was done 2 years ago. Will try to remember what for.